spring-cloud-skipper-shell-commands/src/main/java/org/springframework/cloud/skipper/shell/command/PackageCommands.java

```diff
@@ -49,10 +59,36 @@ public PackageCommands(SkipperClient skipperClient) {
 		this.skipperClient = skipperClient;
 	}
 
-	@ShellMethod(key = "package list", value = "Get the package metadata")
-	public String packageMetadata(
-			@ShellOption(help = "boolean to set for more detailed package metadata", defaultValue = "false") boolean details) {
-		return skipperClient.getPackageMetadata(details);
+	@ShellMethod(key = "package search", value = "Search for the packages")
+	public Table searchPackage(
+			@ShellOption(help = "wildcard expression to search for the package name", defaultValue = NULL) String name,
+			@ShellOption(help = "boolean to set for more detailed package metadata") boolean details)
+			throws Exception {
+		Resources<PackageMetadata> resources = skipperClient.getPackageMetadata(name, details);
+		TableBuilder tableBuilder = null;
+		if (!details) {
+			LinkedHashMap<String, Object> headers = new LinkedHashMap<>();
+			headers.put("name", "Name");
+			headers.put("version", "Version");
+			headers.put("description", "Description");
+			TableModel model = new BeanListTableModel<>(resources.getContent(), headers);
+			tableBuilder = new TableBuilder(model);
+			TableUtils.applyStyle(tableBuilder);
+		}
+		else {
+			ObjectMapper mapper = new ObjectMapper();
+			String[][] data = new String[resources.getContent().size()][1];
+			PackageMetadata[] packageMetadataResources = resources.getContent().toArray(new PackageMetadata[0]);
+			for (int i = 0; i < resources.getContent().size(); i++) {
+				for (int j = 0; j < 1; j++) {
+					data[i][j] = mapper.writeValueAsString(packageMetadataResources[i]);
+				}
+			}
+			TableModel model = new ArrayTableModel(data);
+			// todo: apply style
```

Comment:
Not sure table style is the issue, since the contents are a raw json string.  Modified to print yml, with a separator per record.